Mathlib/Util/Superscript.lean

```diff
@@ -225,8 +231,23 @@ def scriptParser.formatter (name : String) (m : Mapping) (k : SyntaxNodeKind) (p
   | .ok newStack =>
     set { st with stack := stack ++ newStack }
 
+/-- Returns true if every character in `stx : Syntax` can be superscripted
+(or subscripted). -/
+private partial def isValid (m : Mapping) : Syntax → Bool
+  | .node _ kind args => !(scripted kind) && args.all (isValid m)
+  | .atom _ s => valid s
+  | .ident _ _ s _ => valid s.toString
+  | _ => false
+where
+  valid (s : String) : Bool :=
+    s.toList.all fun x ↦ x == ' ' || m.toSpecial.contains x
```

Comment:
```suggestion
    s.all ((m.toSpecial.insert ' ' ' ').contains ·)
```
optional golfing. Chose this since `.insert ' ' ' '` is also used further up in the document